pkg/gardenlet/controller/managedseed/managedseed_controller.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package managedseed
+
+import (
+	"k8s.io/client-go/tools/cache"
+
+	seedmanagementv1alpha1 "github.com/gardener/gardener/pkg/apis/seedmanagement/v1alpha1"
+	"github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/utils"
+)
+
+func (c *Controller) managedSeedAdd(obj interface{}, immediately bool) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		return
+	}
+
+	if immediately {
```

Comment:
OK

Reply:
Actually, after taking a second look, I am not sure what exactly is the issue you have in mind:

* We do want to enqueue new ManagedSeeds with a jittered delay, since when we restart `gardenlet` all existing ManagedSeeds will be detected as "new" and we want to spread their reconciliation over a certain period.
* The `Add` function is already being reused for updates.

From my perspective the current behavior is exactly what we want: "creates" are scheduled with a jittered delay, while "updates" and "deletes" are scheduled immediately. This was also the behavior before, and believe it was on purpose, not a mistake. Am I missing something?

Reply:
OK

Reply:
What you suggested doesn't work, adding such a check in the beginning of `managedSeedAdd` leads to not reconciling managed seed when gardenlet is restarted. I assume this is not what we want.

Instead, I removed the `immediately` parameter and ensured that the managed seed is added to the queue without delay if `managedSeed.Generation != managedSeed.Status.ObservedGeneration` (actual creates or updates), and with delay otherwise (gardenlet restarts). I believe this is what you requested.